src/java/org/apache/cassandra/db/marshal/AbstractType.java

```diff
@@ -669,12 +753,16 @@ public final AbstractType<?> withUpdatedUserTypes(Iterable<UserType> udts)
      */
     public AbstractType<?> expandUserTypes()
     {
-        return this;
+        // Note that UDT type must override this method.
```

Comment:
Maybe we want to add an assertion like the one in `referencesUserType`, or just do nothing, but we should probably use the same criteria.